packages/astro/src/core/routing/match.ts

```diff
@@ -59,7 +59,8 @@ export function isRouteServerIsland(route: RouteData): boolean {
  */
 export function isRequestServerIsland(request: Request, base = ''): boolean {
 	const url = new URL(request.url);
-	const pathname = url.pathname.slice(base.length);
+	const pathname =
+		base == '/' ? url.pathname.slice(base.length) : url.pathname.slice(base.length + 1);
```

Comment:
Does this take into account base with trailing/leading slashes? eg. will it work if base if `/custom`, `custom/` and `/custom/`?

Reply:
It doesn't, and I believe a lot of our code doesn't handle that. Another alternative is to use `.includes` instead of `startsWith`. What do you think?